backend/sources/facebook/sender/src/main/java/co/airy/core/sources/facebook/Sender.java

```diff
@@ -64,27 +97,41 @@ public void startStream() {
 
                             return aggregate;
                         })
-                .join(channelsTable, SendMessageRequest::getChannelId, (aggregate, channel) -> {
+                .join(channelsTable, Conversation::getChannelId, (aggregate, channel) -> {
                     aggregate.setChannel(channel);
                     return aggregate;
                 });
 
+        // Send outbound messages
         messageStream.filter((messageId, message) -> DeliveryState.PENDING.equals(message.getDeliveryState()))
-                .join(contextTable, (message, sendMessageRequest) -> {
-                    sendMessageRequest.setMessage(message);
-                    return sendMessageRequest;
-                })
+                .join(conversationTable, (message, conversation) -> new SendMessageRequest(conversation, message))
                 .mapValues(this::sendMessage)
                 .to(new ApplicationCommunicationMessages().name());
 
+        // Fetch missing metadata
+        conversationTable
+                // To avoid any redundant fetch contact operations the suppression interval should
+                // be higher than the timeout of the Facebook API
+                .suppress(Suppressed.untilTimeLimit(Duration.ofMillis(streams.getSuppressIntervalInMs()), Suppressed.BufferConfig.unbounded()))
```

Comment:
This suppression is optional, but we can use it to minimize the risk of a new conversation requesting the contact data too many times. If we stream the table without suppression this app could be badly behaved and call the Facebook API multiple times when a stream of messages for a new conversation comes in.